collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java

```diff
@@ -565,117 +684,83 @@ else if (columnNames.contains(path)) { // if column present in sheet
         return true;
     }
 
-    private void uploadPhotosToPicasa(HashMap<String, String> photos,
-                                      HashMap<String, PhotoEntry> uploaded, PicasaClient client, AlbumEntry albumToUse,
-                                      File instanceFile) throws IOException {
-        for (String key : photos.keySet()) {
-            String filename = instanceFile.getParentFile() + "/" + photos.get(key);
-            File toUpload = new File(filename);
-
-            // first check the local content provider
-            // to see if this photo already has a picasa_id
-            String selection = Images.Media.DATA + "=?";
-            String[] selectionArgs = {
-                    filename
-            };
-            Cursor c = Collect.getInstance().getContentResolver()
-                    .query(Images.Media.EXTERNAL_CONTENT_URI, null, selection, selectionArgs, null);
-            if (c.getCount() != 1) {
-                c.close();
-                throw new FileNotFoundException(picasa_fail
-                        + Collect.getInstance().getString(R.string.picasa_upload_error, filename));
-            }
-
-            // assume it's not already in picasa
-            boolean inPicasa = false;
-
-            // this will contain the link to the photo
-            PhotoEntry picasaPhoto = null;
-
-            c.moveToFirst();
-            String picasa_id = c.getString(c.getColumnIndex(Images.Media.PICASA_ID));
-            if (picasa_id == null || picasa_id.equalsIgnoreCase("")) {
-                // not in picasa, so continue
-            } else {
-                // got a picasa ID, make sure it exists in this
-                // particular album online
-                // if it does, go on
-                // if it doesn't, upload it and update the picasa_id
-                if (albumToUse.numPhotos != 0) {
-                    PicasaUrl photosUrl = new PicasaUrl(albumToUse.getFeedLink());
-                    AlbumFeed albumFeed = client.executeGetAlbumFeed(photosUrl);
-
-                    for (PhotoEntry photo : albumFeed.photos) {
-                        if (picasa_id.equals(photo.id)) {
-                            // already in picasa, no need to upload
-                            inPicasa = true;
-                            picasaPhoto = photo;
-                        }
-                    }
-                }
-            }
+    private String uploadFileToDrive(HashMap<String, String> photosToUpload, String key, com.google.api.services.drive.model.File folder, File toUpload) throws IOException {
```

Comment:
This parameter list is a bit confusing. There's no reason to pass in the full map of photos to upload since you only need the name of the file. You also don't need a reference to `folder`, only its ID. I think a better set of params would be `File toUpload, String mediaName, String destinationFolderID`